llama_index/llm_predictor/base.py

```diff
@@ -92,6 +98,16 @@ def predict(self, prompt: Prompt, **prompt_args: Any) -> str:
             # NOTE: this is an approximation, only for token counting
             formatted_prompt = messages_to_prompt(messages)
         else:
+            if self.system_prompt:
```

Comment:
Would it make more sense to do this once we get the `formatted_prompt` string? Then there is less fiddling around with templates here 🤔

Reply:
I was considering that, but the query wrapper prompt piece needs to be done before the format, so I wanted to keep the system prompt and query wrapper prompt additions together in the "same step". The reason I did that was to actually put it all into one new function, but then I forgot to actually do that 😅. So I think it makes more sense in my new commit. I could still pull the format into the new function as well, and then do query wrapper -> format -> system all in the new function, if you think it would be better though